presto-main/src/main/java/com/facebook/presto/type/FloatOperators.java

```diff
@@ -70,4 +78,61 @@ public static long negate(@SqlType(StandardTypes.FLOAT) long value)
     {
         return Float.floatToRawIntBits(-Float.intBitsToFloat((int) value));
     }
+
+    @ScalarOperator(EQUAL)
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean equal(@SqlType(StandardTypes.FLOAT) long left, @SqlType(StandardTypes.FLOAT) long right)
+    {
+        return Float.intBitsToFloat((int) left) == Float.intBitsToFloat((int) right);
```

Comment:
0.0f == -0.0f (dual representation of zero in floats)
Bit representation of 0.0f is different than -0.0f but in fact it's the same number and equal(0.0f, -0.0f) should return true according to IEEE754 standard.